### PR TITLE
Cachecontrol

### PR DESCRIPTION
**Pull request by j831**
This adds a Cache-Control expires header if the config setting is set. 

https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching
http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html#ExpirationAddingHeadersInS3

### CODE REVIEW

File: magento/app/code/community/Uaudio/Storage/Model/League/AwsS3Adapter.php [hunk 1]
```diff
@@ -27,6 +27,7 @@ class Uaudio_Storage_Model_League_AwsS3Adapter extends AwsS3Adapter {
     protected function upload($path, $body, Config $config) {
         $key = $this->applyPathPrefix($path);
         $mimetype = MimeType::detectByFileExtension(pathinfo($path, PATHINFO_EXTENSION));
+        if (isset($this->options['cachecontrol'])) $cachecontrol = 'max-age='.$this->options['cachecontrol'];
```
Comment (dmulter) [inline magento/app/code/community/Uaudio/Storage/Model/League/AwsS3Adapter.php]: Shouldn't $cachecontrol clear out any old value if not set?

File: magento/app/code/community/Uaudio/Storage/Model/League/AwsS3Adapter.php [hunk 2]
```diff
@@ -42,6 +43,7 @@ protected function upload($path, $body, Config $config) {
                 'Bucket' => $this->bucket,
                 'CopySource' => $this->bucket.DS.$key,
                 'ContentType' => $mimetype,
+                'CacheControl' => $cachecontrol,
```
Comment (dmulter) [inline magento/app/code/community/Uaudio/Storage/Model/League/AwsS3Adapter.php:46]: What happens if $cachecontrol isn't set?

Comment (j831) [reply magento/app/code/community/Uaudio/Storage/Model/League/AwsS3Adapter.php:46]: S3 ignores an empty value

File: magento/app/code/community/Uaudio/Storage/Model/Storage/S3.php [hunk 3]
```diff
@@ -46,6 +51,7 @@ public function __construct($settings=[]) {
         $this->_region = isset($settings['s3_region'])? $settings['s3_region'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_region');
         $this->_bucket = isset($settings['s3_bucket']) ? $settings['s3_bucket'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_bucket');
         $this->_folder = isset($settings['s3_folder']) ? $settings['s3_folder'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_folder');
+        $this->_cachecontrol = isset($settings['s3_cachecontrol']) ? $settings['s3_cachecontrol'] : Mage::getStoreConfig('system/media_storage_configuration/media_s3_cachecontrol');
```
Comment (schmitty42) [inline magento/app/code/community/Uaudio/Storage/Model/Storage/S3.php:54]: Should maybe validate that the cachecontrol config is an int.

Comment (j831) [reply magento/app/code/community/Uaudio/Storage/Model/Storage/S3.php:54]: it's coming out of magento config... maybe follow this line with        

`if (!is_numeric($this->_cachecontrol)) unset($this->_cachecontrol);
`

Comment (schmitty42) [reply magento/app/code/community/Uaudio/Storage/Model/Storage/S3.php:54]: Or maybe an isset($settings['s3_cachecontrol) && is_numeric($settings['s3_cachecontrol']) ? ...

File: magento/app/code/community/Uaudio/Storage/Model/Storage/S3.php [hunk 4]
```diff
@@ -46,6 +51,8 @@ public function __construct($settings=[]) {
...
+        if (!is_numeric($this->_cachecontrol)) unset($this->_cachecontrol);
```
Comment (j831) [inline magento/app/code/community/Uaudio/Storage/Model/Storage/S3.php:55]: tested with various wonky values and it seems to work as intended

**Pull request merged**